.pipelines/templates/rp-dev/template-generate-rp-dev-config.yml

```diff
@@ -0,0 +1,44 @@
+parameters:
+  secretSA: ""
+  location: ""
+  resourceGroup: ""
+  databaseAccountName: ""
+  keyvaultPrefix: ""
+  aroImage: ""
+
+steps:
+  - script: |
+      set -e
+      # Export the SECRET_SA_ACCOUNT_NAME environment variable and run make secrets
+      export SECRET_SA_ACCOUNT_NAME=${{ parameters.secretSA }} && make secrets
+
+      # Define the expected directory and file names
+      expected_dir="secrets"
+      files=("env" "dev-ca.crt" "dev-client.crt")
+
+      # Validate that the secrets directory and required files exist
+      [ -d "$expected_dir" ] || { echo "Directory '$expected_dir' has not been created."; exit 1; }
+      for file in "${files[@]}"; do
+        [ -f "$expected_dir/$file" ] || { echo "File '$file' does not exist inside the directory '$expected_dir'."; exit 1; }
+      done
+    name: download-secrets
+    displayName: 🔑 Downloading certificates and secrets from storage account
+
+  - script: |
+      set -e
+      # Export environment variables from parameters
+      export LOCATION=${{ parameters.location }} RESOURCEGROUP=${{ parameters.resourceGroup }} DATABASE_ACCOUNT_NAME=${{ parameters.databaseAccountName }}
+      export KEYVAULT_PREFIX=${{ parameters.keyvaultPrefix }} ARO_IMAGE=${{ parameters.aroImage }}
+      # Source environment variables from the secrets file
+      source secrets/env
+
+      # Generate SSH key
+      ssh-keygen -t rsa -N "" -f ~/.ssh/id_rsa
```

Comment:
I think this key is being used to ssh to RP/Gateway right?  I don't see this being exported for SRE use outside the pipeline so we wouldn't have ssh access to the RP/Gateway.  We could add a new ssh pub/priv key to `rharosecretsdev` and use that here instead, that would give everyone access to ssh to the RP/Gateway VMs.

Reply:
Gotcha sounds like you just resolved @SudoBrendan [question](https://github.com/Azure/ARO-RP/pull/3670#discussion_r1678286758) :)
IIUC then your suggestion is something that should be done with shared RP creation? If so, then it isn't something we should do in this PR

Reply:
@tiguelu is correct.  We can do this in a few steps:
1. Make sure you have the latest secrets `SECRET_SA_ACCOUNT_NAME=rharosecretsdev make secrets`
2. Outside of the pipeline create a SSH priv/pub key (one time only or any time to renew / rotate the keys)
3. Move the public / private key to `secrets/`
4. Update `secrets/env` file with:
```
export SSH_PRIVATE_KEY=secrets/<whatever the Priv key file path is>
export SSH_PUBLIC_KEY=secrets/<whatever the pub key file path is>
```
5. Run `SECRET_SA_ACCOUNT_NAME=rharosecretsdev make secrets-update`, this will update the storage blob, overwriting the previous version so use caution :).

You should be able to remove `ssh-keygen ...` from the pipeline and it should just work, `make dev-config.yaml` should see the `SSH_PUBLIC_KEY` environment variable and

Reply:
SGTM I will test in a follow-up, thank you @tony-schndr!